File: Project/Backend/src/ArtAuction.Application/Services/PostSoldServices.cs
using ArtAuction.Application.DTOs.PostSold;
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Application.Interfaces.Services;
using AutoMapper;

namespace ArtAuction.Application.Services;

public class PostSoldServices : IPostSoldServices
{
    // Attributes
    private readonly IMapper _mapper;
    private readonly IPostSoldRepo _postSoldRepo;
    
    // Constructor
    public PostSoldServices(IMapper mapper, IPostSoldRepo postSoldRepo)
    {
        this._mapper = mapper;
        this._postSoldRepo = postSoldRepo;
    }
    
    // Methods
    public async Task<ICollection<PostSoldDto>> GetAllPostSolds()
    {
        // Get post sold data
        var postSolds = await _postSoldRepo.GetAllPostSolds();
        
        // Return mapped DTO
        return _mapper.Map<ICollection<PostSoldDto>>(postSolds);
    }

    public async Task<UnpaidPostSoldDto> GetUnpaidPostSoldForBuyer(int buyerId)
    {
        // Get unpaid post for buyer
        var unpaidPost = await _postSoldRepo.GetUnpaidPostSoldByBuyer(buyerId);
        
        // Return mapped DTO
        return _mapper.Map<UnpaidPostSoldDto>(unpaidPost);
    }

    public async Task<ICollection<BuyerPostSoldDto>> GetBuyerPostSolds(int buyerId)
    {
        // Get post sold for buyer
        var postSolds = await _postSoldRepo.GetBuyerPostSolds(buyerId);
        
        // Return mapped DTO
        return _mapper.Map<ICollection<BuyerPostSoldDto>>(postSolds);
    }

    public async Task<bool> MarkAsPaid(PostSoldPaidDto postSoldPaidDto)
    {
        // Mark post sold record as paid
        return await _postSoldRepo.MarkAsPaid(postSoldPaidDto.ArtworkPostId, postSoldPaidDto.BuyerId);
    }
}

File: AuctionServices/src/main/java/com/artauction/auctionservice/Services/Interfaces/PostSoldServices.java
package com.artauction.auctionservice.Services.Interfaces;

import com.artauction.auctionservice.Dtos.PostSold.BuyerPostSoldDto;
import com.artauction.auctionservice.Dtos.PostSold.PostSoldDto;
import com.artauction.auctionservice.Dtos.PostSold.PostSoldPaidDto;
import com.artauction.auctionservice.Dtos.PostSold.UnpaidPostSoldDto;

import java.util.Collection;
import java.util.List;
import java.util.Optional;

public interface PostSoldServices {
    Collection<PostSoldDto> getAllPostSolds() throws Exception;
    List<UnpaidPostSoldDto> getUnpaidPostSoldForBuyer(int buyerId);
    Collection<BuyerPostSoldDto> getBuyerPostSolds(int buyerId);
    boolean markAsPaid(PostSoldPaidDto postSoldPaidDto);
}
